src/dwalk/dwalk.c

```diff
@@ -62,71 +114,83 @@ static int print_flist_distribution(struct distribute_option *option, mfu_flist*
                  /* found the bin set bin index & increment its count */
                  dist[j]++;
- 
-                 /* a file for this bin was found so can't belong to 
+
+                 /* a file for this bin was found so can't belong to
                   * last bin (so set the flag) & exit the loop */
                  max_bin_flag = 1;
                  break;
-             }       
+             }
          }
- 
+
          /* if max_bin_flag is still -1 then the file belongs to the last bin */
          if (max_bin_flag < 0) {
              dist[separators]++;
          }
     }
- 
+
     /* get the total sum across all of the bins */
     uint64_t* disttotal = (uint64_t*) MFU_MALLOC((separators + 1) * sizeof(uint64_t));
     MPI_Allreduce(dist, disttotal, (uint64_t)separators + 1, MPI_UINT64_T, MPI_SUM, MPI_COMM_WORLD);
- 
+
     /* Print the file distribution */
     if (rank == 0) {
          /* number of files in a bin */
          uint64_t number;
-         printf("Range\tNumber\n");
-         for (int i = 0; i <= option->separator_number; i++) {
-             printf("[");
+         double size_tmp;
+         const char* size_units;
+         printf("%-49s %-10s\n", "Range", "Number");
+         for (int i = 0; i <= separators; i++) {
+             printf("%-10s", "[");
              if (i == 0) {
-                 printf("0");
+                 printf("%-7.3lf %-10s", 0.000, "B");
              } else {
-                 printf("%"PRIu64, option->separators[i - 1] + 1);
+                 mfu_format_bytes((uint64_t)option->separators[i - 1],
+                                  &size_tmp, &size_units);
+                 printf("%-7.3lf %-10s", size_tmp, size_units);
              }
- 
-             printf("~");
- 
-             if (i == option->separator_number) {
-                 number = disttotal[i];
-                 printf("MAX]\t%"PRIu64"\n", number);
+
+             printf("%-10s", " - ");
+
+             if (file_histogram) {
+                mfu_format_bytes((uint64_t)option->separators[i],
+                                 &size_tmp, &size_units);
+                number = disttotal[i];
+                mfu_format_bytes((uint64_t)option->separators[i],
+                                 &size_tmp, &size_units);
+                printf("%-7.3lf %-2s) %-3"PRIu64"\n", size_tmp,
+                       size_units, number);
              } else {
-                 number = disttotal[i];
-                 printf("%"PRIu64"]\t%"PRIu64"\n",
-                 option->separators[i], number);
+                if (i == separators) {
+                    number = disttotal[i];
+                    printf("%-10s) %-3"PRIu64"\n", "MAX", number);
+                } else {
+                    number = disttotal[i];
+                    mfu_format_bytes((uint64_t)option->separators[i],
+                              &size_tmp, &size_units);
+                    printf("%-7.3lf %-2s) %-3"PRIu64"\n", size_tmp, size_units, number);
```

Comment:
Does setting this to 3 truncate numbers longer than 3 characters?  Anyway, I don't think we need to limit it, so we could just drop the 3.

Reply:
I see there are examples with 4 characters in the numbers, so I guess it doesn't limit it.  Let's drop it anyway, though, unless it's there for a reason.